crates/storage/src/lib.rs

```diff
@@ -272,6 +299,9 @@ pub trait FileReader: Send + Sync + HasFileId {
     /// deleted on drop.
     fn mark_for_checkpoint(&self);
 
+    /// Commits the file to stable storage.
+    fn commit(&self) -> Result<(), StorageError>;
```

Comment:
This I don't understand, if it's a reader, what is it committing?

Reply:
When we finish writing a file to storage, we transform the FileWriter into a FileReader. At this point, the data that was written may not be committed to stable storage. The handle we have available at that point is a FileReader, so that seems like the appropriate handle to use for committing.

We could, I suppose, invent a new handle called, for example, a FileCommitter, and return both a FileReader and a FileCommitter when completing a FileWriter. I guess there could be a small benefit in clarity there, but it seems to me like there's an increase in complexity.

FWIW, fsync doesn't require the file to be opened for write.

Reply:
I added a commit that separates the semantics into FileCommitter and FileReader without needing multiple handles. Take a look, I'm pretty happy with it.